test/parallel/test-domain-abort-when-disposed.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const common = require('../common');
+const JSStream = process.binding('js_stream').JSStream;
```

Comment:
could you put this together with `Socket` and comment something like `// arbitrary objects that trigger async_hooks`